### ISSUE TITLE
I want to define the order when opening multiple modals

### ISSUE THREAD

**Issue by okayu8** [cloudle/ruui#25]
When I click the button of modal 1, I try to open a modal 2 on top of it.
Modal 2 does not mean you can only open from modal 1. You can also open from another location.
After opening modal 2 in another place and opening modal 2 from modal 1, modal 2 hides behind modal 1.

I think that this is because the order of modals [expanded](url) by map changes.
Is it possible to define the order of the modal array by property etc.?

**Comment by cloudle** [comment]
@okayu8 yep, modals list now stored in a javascript HashMap, eg:
```
modalState = {
  default: {...},
  namedModal: {...}
}
```
and we render it back using `Object.keys(modalState).forEach(...)`.
Since javascript sort `HashMap` by `key` => `default` stand before `namedModal`, for example.. if we add a modal by this code:
```
store.dispatch(ruuiActions.toggleModal(true, { id: 'aModal', component: MyModal }));
```
it update our HashMap state to:
```
modalState = {
  aModal: {...},
  default: {...},
  namedModal: {...}
}
```

which cause our rendering put "aModal" under other modals => what we could do to bring it on top is change our code to:
```
store.dispatch(ruuiActions.toggleModal(true, { id: 'zModal', component: MyModal }));
```
which make the state like this:
```
modalState = {
  default: {...},
  namedModal: {...},
  zModal: {...},
}
```

In summary - for now, please use `key` to handle layer priority ;)
I'll allow passing an option e.g `zIndex` when toggleModal, and let you know when it is ready.

**Comment by okayu8** [comment]
@cloudle Sorry for my late reply.
I was able to solve with your answer.
